karate-core/src/main/java/com/intuit/karate/core/ScenarioRuntime.java

```diff
@@ -105,6 +106,11 @@ public boolean isFailed() {
         return error != null || result.isFailed();
     }
 
+    public boolean hasIgnoredErrors() {
+        return ignoringFailureSteps; // result.getStepResults().stream().anyMatch(StepResult::isErrorIgnored);
+        // to review in code review if boolean is required, avoiding stream() might be desirable for performance
```

Comment:
I think stream is fine. it applies only if errors right ? also from experience when I tried premature optimization like this maintaining state it bit me in unexpected places later. e.g. now all the result conversion to JSON is done on demand, no caching of errors and counts

Reply:
The challenge is the code I have on line 467 of the same file meant to "reset".
- If the engine is no longer ignoring errors we either want to set the engine failure reason to null (to continue executing steps) or stop execution.

When this feature is not in use we'd always be looking back at all the steps to check whether any was marked as error ignored for no reason and might be impactful if tests are long.

I'd also like to avoid maintaining the state in the Engine as that gets propagated to called features (I'll comment on your other comment to explain why I would prefer to avoid this).

Let me know whether you can think of something else, I'll review the rest of the PR and maybe have other ideas for this.

Reply:
ok ignore my comment if this is state in runtime / engine